server/channels/app/content_flagging_setup.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"sync"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/pkg/errors"
+)
+
+const contentFlaggingSetupDoneKey = "content_flagging_setup_done"
+
+var contentFlaggingGroupID string
+var contentFlaggingSetupOnce sync.Once
+
+func (a *App) GetContentFlaggingGroupID() (string, error) {
+	if contentFlaggingGroupID != "" {
+		return contentFlaggingGroupID, nil
+	}
+
+	group, err := a.Srv().propertyService.RegisterPropertyGroup(model.ContentFlaggingGroupName)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to register Content Flagging group")
+	}
+	contentFlaggingGroupID = group.ID
+
+	return contentFlaggingGroupID, nil
+}
+
+func (a *App) SetupContentFlaggingProperties() error {
+	var setupErr error
+
+	contentFlaggingSetupOnce.Do(func() {
+		groupId, err := a.GetContentFlaggingGroupID()
+		if err != nil {
+			setupErr = errors.Wrap(err, "failed to get Content Flagging group ID when setting up content flagging properties")
+			return
+		}
+
+		if system, err := a.Srv().Store().System().GetByName(contentFlaggingSetupDoneKey); err == nil && system.Value == "true" {
+			// If the setup is already done, we skip the property creation
+			return
+		}
+
+		// register status property
+		properties := []*model.PropertyField{
+			{
+				GroupID: groupId,
+				Name:    "status",
```

Comment:
Maybe it's worth defining these as constants. WDYT?

Reply:
Done.